livestream-compose-sample/src/main/java/io/getstream/livestream/compose/ui/LiveStreamChannels.kt

```diff
@@ -0,0 +1,164 @@
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.ViewStream
+import androidx.compose.material.ripple.rememberRipple
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.viewmodel.channel.ChannelListViewModel
+import io.getstream.livestream.compose.R
+import io.getstream.livestream.compose.models.LiveStreamChannelItem
+import io.getstream.livestream.compose.models.LiveStreamType
+import io.getstream.livestream.compose.models.LiveStreamType.*
+import io.getstream.livestream.compose.randomArtWork
+import io.getstream.livestream.compose.randomDescription
+
+/**
+ * View component to add custom live stream channels screen
+ *
+// * @param context - Local context for fetching resources to draw Icons or vectors
+ * @param channelListViewModel - Stream channel list ViewModel to bind all channel data to custom UI provided here.
+ * @param modifier - Modifier for styling.
+ * @param isDarkTheme - a Boolean state to get Theme is dark or light , used for deciding card background
+ * @param columnCount - a toggle State to switch between 2 column grid to single column list
+ */
+@ExperimentalMaterialApi
+@ExperimentalFoundationApi
+@Composable
+fun LiveStreamChannels(
+    channelListViewModel: ChannelListViewModel,
+    modifier: Modifier = Modifier,
+    isDarkTheme: Boolean = false,
+    columnCount: Int = 2
+) {
+    LaunchedEffect(Unit) {
+        channelListViewModel.start()
+    }
+    val cardBackground = if (isDarkTheme)
+        colorResource(id = R.color.cardview_dark_background)
+    else
+        colorResource(id = R.color.cardview_light_background)
+    val context: Context = LocalContext.current
+
+    val channels = channelListViewModel.channelsState.channels.map { channel ->
+        LiveStreamChannelItem(
+            channelId = channel.cid,
+            channelArt = context.randomArtWork(),
+            channelName = channel.name,
+            channelDescription = context.randomDescription()
+        )
+    }
+
+    LazyVerticalGrid(
+        modifier = modifier,
+        cells = GridCells.Fixed(columnCount),
+        contentPadding = PaddingValues(bottom = 12.dp, top = 12.dp),
+        content = {
+            items(channelListViewModel.channelsState.channels.size) { index ->
+                Card(
```

Comment:
Pull this out to a component anyway, it's easier to inspect this component if we don't have 50 lines of code for a component. We need to decouple components into small items, to be easily readable and scannable.

Reply:
Refactored to new file